fil_fungible_token/src/token/mod.rs

```diff
@@ -362,8 +366,8 @@ where
             )));
         }
 
-        // owner and operator must exist to burn from
-        let operator = expect_id(operator)?;
+        // instantiate the actors if required
+        let operator = self.resolve_to_id(operator)?;
         let owner = self.resolve_to_id(owner)?;
```

Comment:
Here in the `burn` method, an argument could be made for skipping account initialization when handling zero-amounts. 

If the both `operator` and `owner` are uninitialised but `owner` is an Actor-address, the `operator` address will be initialised whilst the `burn` method fails. In some implementations, our consumers may catch this error and not abort the entire VM-level transaction. This may lead to unexpected state being persisted on-chain.